cli/src/main/java/com/devonfw/tools/ide/tool/LocalToolCommandlet.java

```diff
@@ -323,127 +318,66 @@ private ToolInstallation createToolInstallation(Path rootDir, VersionIdentifier
     return createToolInstallation(rootDir, resolvedVersion, toolVersionFile, false);
   }
 
-  @Override
-  public void runTool(ProcessMode processMode, VersionIdentifier toolVersion, String... args) {
-
-    Path binaryPath;
-    Path toolPath = Path.of(getBinaryName());
-    if (toolVersion == null) {
-      install(true);
-      binaryPath = toolPath;
-    } else {
-      throw new UnsupportedOperationException("Not yet implemented!");
-    }
-
-    if (Files.exists(this.dependency.getDependencyJsonPath(getConfiguredEdition()))) {
-      setDependencyRepository(getInstalledVersion());
-    } else {
-      this.context.trace("No Dependencies file found");
-    }
-
-    ProcessContext pc = this.context.newProcess().errorHandling(ProcessErrorHandling.WARNING).executable(binaryPath).addArgs(args);
-
-    for (String key : this.dependenciesEnvVariablePaths.keySet()) {
+  public void setEnvironment(EnvironmentContext context, Path dependencyPath, String dependencyName) {
 
-      String dependencyPath = this.dependenciesEnvVariablePaths.get(key);
-      pc = pc.withEnvVar(key, dependencyPath);
-    }
-
-    pc.run(processMode);
+    String pathVariable = dependencyName.toUpperCase(Locale.ROOT) + "_HOME";
+    context.withEnvVar(pathVariable, dependencyPath.toString());
```

Comment:
IMHO you are missing the `null` check here since by default `null` is passed as `EnvironmentContext` but maybe it does not need to go here but in all places calling this method.

Reply:
Now `setEnvironment` does the `null` check so all perfect and this can be resolved.